site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -63,11 +63,96 @@ public function RandomizePeers($gradeable_id) {
         5 - Shift the random order by the offsets to create the matrix, with no duplicates, and exactly Y assignments and & graders for each student.  no student grades self.
         */
         $number_to_grade = $_POST['number_to_grade'];
+        $restrict_to_registration = $_POST['restrict_to_registration'];
         $gradeable = $this->tryGetGradeable($gradeable_id);
         if ($gradeable === false) {
             $this->core->addErrorMessage('Invalid Gradeable!');
             $this->core->redirect($this->core->buildCourseUrl());
         }
+        /* If Restrict to Registration checkbox is checked, then the randomised peer assignments should be restricted to each registration section" */
+        if ($restrict_to_registration == "checked") {
+            $sections = $this->core->getQueries()->getRegistrationSections();
+            foreach ($sections as $i => $section) {
+                $sections[$i] = $section['sections_registration_id'];
+            }
+            /* Duplicating the entire functionality of randomization, but for each registration section rather than whole class" */
```

Comment:
Helper function was always an option but there are many boolean(s) here
1 -  All grade all when Restricted to registration section
2 -  All grade all when Restricted to Submission before Grading
3 - All grade all when restricted to registration section an also submission before grading
4 - All grades all when no options are selected
I thought it will only make everything more complex so thought of doing everything with same set of variables inside one function (with proper comments) 
If you ask me to, I can look for a good way to implement is using a helper function.

Reply:
Again, don't let it hold you up. The code still works (: